cube_holder.scad

```diff
@@ -17,29 +17,39 @@ cube_holder();
 // Begin modules that can be used to assemble this piece in solver_3.scad
 
 module above_cube_holder() {
-  up(thickness/2 + servo_gear_thickness) above_frame() children();
+  up(servo_gear_thickness) above_frame() children();
 }
 
 module cube_holder() {
+  up(servo_gear_thickness) onto_point()
+    piece_sleeve();
+  holder_gear();
+}
+
+module piece_sleeve() {
   piece_size = cube_size / 3 + margin;
   holder_open_size = piece_size + thickness;
   holder_closed_size = piece_size + thickness + embedded_thickness;
-  difference() {
-    up(servo_gear_thickness - thickness/2)
-      onto_point(holder_closed_size)
-      difference() {
+  translate([-piece_size/2,piece_size/2,piece_size/2]) {
+    difference() {
+      translate([thickness/2, (embedded_thickness-thickness)/2, (embedded_thickness-thickness)/2])
+        ccube([holder_open_size, holder_closed_size, holder_closed_size], x+y+z);
+      rounded_cube(piece_size, cube_rounding, x+y+z);
+      translate([-thickness, 0, 0]) rounded_cube(piece_size, cube_rounding, x+y+z);
```

Comment:
these should get comments. I know, I wrote it, and I totally had to open the model and play with these values to understand what was going on. I should have commented it originally!

Reply:
Done!